### PR TITLE
feat: update setGiteaOIDCConfig function and logic

### PR DESCRIPTION
**Pull request by ferruhcihan**
### Description:
This PR fixes/updates  Gitea CLI command to create/update the Gitea OIDC configuration.
Is paired with: https://github.com/linode/apl-core/pull/1673

### CODE REVIEW

File: src/operator/gitea.ts [hunk 1]
```diff
@@ -480,11 +480,18 @@ async function setGiteaOIDCConfig() {
     const execCommand = [
       'sh',
       '-c',
-      `AUTH_ID=$(gitea admin auth list --vertical-bars | grep -E "\\|otomi-idp\\s+\\|" | grep -iE "\\|OAuth2\\s+\\|" | awk -F " " '{print $1}' | tr -d '\\n') && gitea admin auth update-oauth --id "$AUTH_ID" --key "${clientID}" --secret "${clientSecret}" --auto-discover-url "${discoveryURL}"`,
+      `
+      AUTH_ID=$(gitea admin auth list --vertical-bars | grep -E "\\|otomi-idp\\s+\\|" | grep -iE "\\|OAuth2\\s+\\|" | awk -F " " '{print $1}' | tr -d '\\n')
```
Comment (j-zimnowoda) [inline src/operator/gitea.ts:484]: It works. However you can't see it in logs.

**Review by j-zimnowoda — DISMISSED** [review]

File: src/operator/gitea.ts (hunk 1 above)
Comment (j-zimnowoda) [inline src/operator/gitea.ts:484]: This operator should periodically check if oauth setting are in place.

Consider the following use cases:
1. The gitea and gitea-operator are in the running state. However the DNS is not ready yet. What will happen then?
2. Someone remove otomi-idp from Gitea. The operator will recreated but only if it gets restarted

File: src/operator/gitea.ts [hunk 2]
```diff
@@ -130,10 +130,29 @@ const secretsAndConfigmapsCallback = async (e: any) => {
...
+    if (intervalId === null) {
+      intervalId = setInterval(() => {
+        setGiteaOIDCConfig()
+          .catch((error) => {
+            console.error('Error occurred during setGiteaOIDCConfig execution:', error)
+          })
+          .finally(() => {
+            intervalId = null
+          })
+      }, 30 * 1000)
```
Comment (j-zimnowoda) [inline src/operator/gitea.ts:145]: the interval should be configurable via ENV

File: src/operator/gitea.ts (hunk 1 above)
Comment (j-zimnowoda) [reply src/operator/gitea.ts:504]: Thanks for implementing retry mechanism

File: src/operator/gitea.ts [hunk 3]
```diff
@@ -468,44 +487,60 @@ async function setGiteaGroupMapping(podNamespace: string, podName: string) {
...
       '-c',
-      `AUTH_ID=$(gitea admin auth list --vertical-bars | grep -E "\\|otomi-idp\\s+\\|" | grep -iE "\\|OAuth2\\s+\\|" | awk -F " " '{print $1}' | tr -d '\\n') && gitea admin auth update-oauth --id "$AUTH_ID" --key "${clientID}" --secret "${clientSecret}" --auto-discover-url "${discoveryURL}"`,
+      `
+      AUTH_ID=$(gitea admin auth list --vertical-bars | grep -E "\\|otomi-idp\\s+\\|" | grep -iE "\\|OAuth2\\s+\\|" | awk -F " " '{print $1}' | tr -d '\\n')
+      if [ -z "$AUTH_ID" ]; then
+        echo "Gitea OIDC config not found. Adding OIDC config for otomi-idp."
+        gitea admin auth add-oauth --name "otomi-idp" --key "${clientID}" --secret "${clientSecret}" --auto-discover-url "${discoveryURL}" --provider "openidConnect" --admin-group "team-admin" --group-claim-name "groups" --group-team-map '${teamNamespaceString}'
+      elif ${update}; then
+        echo "Gitea OIDC config is different. Updating OIDC config for otomi-idp."
+        gitea admin auth update-oauth --id "$AUTH_ID" --key "${clientID}" --secret "${clientSecret}" --auto-discover-url "${discoveryURL}"
```
Comment (j-zimnowoda) [inline src/operator/gitea.ts:510]: why on update you do not use the `--group-claim-name "groups" --group-team-map '${teamNamespaceString}'` arguments

Comment (j-zimnowoda) [reply src/operator/gitea.ts]: thx

**Review by j-zimnowoda — APPROVED** [review]

File: src/operator/gitea.ts (hunk 2 above)
Comment (ferruhcihan) [reply src/operator/gitea.ts]: ENV added for the interval

**Pull request merged**